### PR TITLE
docs: sto entity comments

### PR DESCRIPTION
**Pull request by victorwiebe**

### CODE REVIEW

File: src/entities/Sto.ts [hunk 1]
```diff
@@ -12,35 +12,85 @@ import {
...
   isPaused: boolean;
+  /**
+   * whether or not the cap has been reached for the sto
+   */
   capReached: boolean;
+  /**
+   * whether or not the sto has been finalized
+   */
   isFinalized: boolean;
   preIssueAllowed: boolean;
```
Comment (shuffledex) [inline src/entities/Sto.ts]: preIssueAllowed is not documented

File: src/entities/Sto.ts [hunk 2]
```diff
@@ -12,35 +12,85 @@ import {
...
   beneficialInvestmentsAllowed: boolean;
```
Comment (shuffledex) [inline src/entities/Sto.ts]: beneficialInvestmentsAllowed is not documented

File: src/entities/Sto.ts [hunk 3]
```diff
@@ -80,6 +130,11 @@ export abstract class Sto<P> extends Entity<P> {
 
   protected context: Context;
 
+  /**
+   * Unserialize string to a Security Token object representation
```
Comment (shuffledex) [inline src/entities/Sto.ts]: Security Token Offering object....

File: src/entities/Sto.ts [hunk 4]
```diff
@@ -262,10 +320,10 @@ export abstract class Sto<P> extends Entity<P> {
   };
 
   /**
-   * Removes a role from a delegate
+   * Remove a role from a delegate
    *
-   * @param delegateAddress wallet address of the delegate
-   * @param role role to revoke
+   * @param args.delegateAddress - wallet address of the delegate
+   * @param args.role - role to revoke
```
Comment (shuffledex) [inline src/entities/Sto.ts]: "Arg properties which are complex interfaces defined elsewhere should not be broken down into their own properties (the autogen tool provides a link to the corresponding interface)"

Comment (victorwiebe) [reply src/entities/Sto.ts]: Not sure I follow, where elsewhere are these defined?

Comment (shuffledex) [reply src/entities/Sto.ts]: StoRole is an enum type, my bad.

File: src/entities/TieredSto.ts [hunk 5]
```diff
@@ -18,39 +18,102 @@ import { Investment } from './Investment';
...
+   */
   currency: Currency;
+  /**
+   * optional beneficiary address to send beneficial investments to
+   */
   beneficiary?: string;
 }
 
+/**
+ * Represents a tiered sto raising in stable coin with a stable coin address
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: `@hidden`

File: src/entities/TieredSto.ts [hunk 6]
```diff
@@ -18,39 +18,102 @@ import { Investment } from './Investment';
...
+   */
   currency: Currency.StableCoin;
+  /**
+   * ethereum address for the stable coin in which funds will be raised
+   */
   stableCoinAddress: string;
 }
 
+/**
+ * Represents a tiered sto where funds are raised in other fund raise types
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: `@hidden`

File: src/entities/TieredSto.ts [hunk 7]
```diff
@@ -66,6 +129,11 @@ export class TieredSto extends Sto<Params> {
 
   public tiers: Tier[];
 
+  /**
+   * Create a new tiered sto instance
+   * @param params parameters for an sto and unique identifiers
+   * @param context the sdk is being used in
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: No need to specify constructor parameters

File: src/entities/TieredSto.ts [hunk 8]
```diff
@@ -166,23 +234,23 @@ export class TieredSto extends Sto<Params> {
...
-   * @param unsoldTokensWallet wallet address that will receive unsold tokens when the end date is reached
-   * @param stableCoinAddresses addresses of supported stablecoins
-   * @param customCurrency custom currency data. Allows the STO to raise funds pegged to a different currency. Optional, defaults to USD
-   * @param customCurrency.currencySymbol symbol of the custom currency (USD, CAD, EUR, etc. Default is USD)
-   * @param customCurrency.ethOracleAddress address of the oracle that states the price of ETH in the custom currency. Only required if raising funds in ETH
-   * @param customCurrency.polyOracleAddress address of the oracle that states the price of POLY in the custom currency. Only required if raising funds in POLY
+   * @param args.startDate - date when the STO should start
+   * @param args.endDate - date when the STO should end
+   * @param args.tiers - tier information
+   * @param args.tiers[].tokensOnSale - amount of tokens to be sold on that tier
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: From the guide: "Arg properties which are complex interfaces defined elsewhere should not be broken down into their own properties (the autogen tool provides a link to the corresponding interface)"

Comment (monitz87) [reply src/entities/TieredSto.ts]: This applies to all the properties within `args.tiers`

File: src/entities/TieredSto.ts [hunk 9]
```diff
@@ -166,23 +234,23 @@ export class TieredSto extends Sto<Params> {
...
+   * @param args.tiers[].tokensWithDiscount - amount of tokens to be sold on that tier at a discount if paid in POLY (must be less than tokensOnSale, defaults to 0)
+   * @param args.tiers[].discountedPrice - price of discounted tokens on that tier (defaults to 0)
+   * @param args.nonAccreditedInvestmentLimit - maximum investment for non-accredited investors
+   * @param args.minimumInvestment - minimum investment amount
+   * @param args.fundraiseCurrencies - array of currencies in which the funds will be raised (ETH, POLY, StableCoin)
+   * @param args.raisedFundsWallet - wallet address that will receive the funds that are being raised
+   * @param args.unsoldTokensWallet - wallet address that will receive unsold tokens when the end date is reached
+   * @param args.stableCoinAddresses - addresses of supported stablecoins
+   * @param args.customCurrency - custom currency data. Allows the STO to raise funds pegged to a different currency. Optional, defaults to USD
+   * @param args.customCurrency.currencySymbol - symbol of the custom currency (USD, CAD, EUR, etc. Default is USD)
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: From the guide: "Arg properties which are complex interfaces defined elsewhere should not be broken down into their own properties (the autogen tool provides a link to the corresponding interface)"

This applies to all the properties within args.customCurrency

File: src/entities/TieredSto.ts [hunk 10]
```diff
@@ -250,6 +321,9 @@ export class TieredSto extends Sto<Params> {
     };
   }
 
+  /**
+   * Hydrating the entity
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: Hydrate the entity

File: src/entities/factories/TieredStoFactory.ts [hunk 11]
```diff
@@ -136,6 +139,11 @@ export class TieredStoFactory extends Factory<TieredSto, Params, UniqueIdentifie
     };
   };
 
+  /**
+   * Creates an instance of the tiered sto factory
```
Comment (monitz87) [inline src/entities/factories/TieredStoFactory.ts]: Create an ...

File: src/entities/factories/TieredStoFactory.ts [hunk 12]
```diff
@@ -136,6 +139,11 @@ export class TieredStoFactory extends Factory<TieredSto, Params, UniqueIdentifie
...
+   *
+   * @param context the context in which sdk will be used
```
Comment (monitz87) [inline src/entities/factories/TieredStoFactory.ts]: No need to specify constructor params

File: src/entities/factories/SimpleStoFactory.ts [hunk 13]
```diff
@@ -75,6 +78,11 @@ export class SimpleStoFactory extends Factory<SimpleSto, Params, UniqueIdentifie
     };
   };
 
+  /**
+   * Creates an instance of the simple sto factory
```
Comment (monitz87) [inline src/entities/factories/SimpleStoFactory.ts]: Create an ...

File: src/entities/factories/SimpleStoFactory.ts [hunk 14]
```diff
@@ -75,6 +78,11 @@ export class SimpleStoFactory extends Factory<SimpleSto, Params, UniqueIdentifie
...
+   *
+   * @param context the context in which sdk will be used
```
Comment (monitz87) [inline src/entities/factories/SimpleStoFactory.ts]: No need to specify constructor params

File: src/entities/TieredSto.ts (hunk 8 above)
Comment (victorwiebe) [reply src/entities/TieredSto.ts]: Great. I will make sure that the interfaces defined elsewhere are documented correctly as I go

File: src/entities/SimpleSto.ts [hunk 15]
```diff
@@ -45,16 +42,23 @@ export class SimpleSto extends Sto<Params> {
     });
   }
 
+  /**
+   * Unique generated Tiered STO id
```
Comment (monitz87) [inline src/entities/SimpleSto.ts]: Parameter and property descriptions should start with a lower-case letter

File: src/entities/Sto.ts [hunk 16]
```diff
@@ -98,40 +96,82 @@ export interface Params {
  * Abstract class used as a base to manage sto functionalities
  */
 export abstract class Sto<P> extends Entity<P> {
+  /**
+   * Uniquely generated id for the STO
```
Comment (monitz87) [inline src/entities/Sto.ts]: Parameter and property descriptions should start with a lower-case letter

File: src/entities/Sto.ts [hunk 17]
```diff
@@ -98,40 +96,82 @@ export interface Params {
...
+   * Type of STO setup
+   */
   public stoType: StoType;
 
   public startDate: Date;
 
   public endDate: Date;
 
+  /**
+   * Wallet where funds raised will be forwarded to
```
Comment (monitz87) [inline src/entities/Sto.ts]: wallet where raised funds...

File: src/entities/Sto.ts [hunk 18]
```diff
@@ -98,40 +96,82 @@ export interface Params {
...
+   */
   public soldTokensAmount: BigNumber;
 
+  /**
+   * Number of investors that have purchased tokens in the STO
+   */
   public investorCount: number;
 
+  /**
+   * Valid currencies that funds can be raised in
```
Comment (monitz87) [inline src/entities/Sto.ts]: Types of currency in which funds can be raised

File: src/entities/Sto.ts [hunk 19]
```diff
@@ -98,40 +96,82 @@ export interface Params {
...
+   */
   public capReached: boolean;
 
+  /**
+   * Whether the STO has been finalized or not
+   */
   public isFinalized: boolean;
 
+  /**
+   * Whether the preissuing of tokens is allowed or not
```
Comment (monitz87) [inline src/entities/Sto.ts]: whether all tokens due to be sold are issued when the STO starts. If false, the appropriate amount of tokens are issued to the buyer whenever a sale is made

File: src/entities/TieredSto.ts [hunk 20]
```diff
@@ -123,16 +120,23 @@ export class TieredSto extends Sto<Params> {
     });
   }
 
+  /**
+   * Unique generated Tiered STO id
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: Parameter and property descriptions should start with a lower-case letter

File: src/types/index.ts [hunk 21]
```diff
@@ -376,9 +376,21 @@ export interface StoTier {
   discountedPrice?: BigNumber;
 }
 
+/**
+ * Unique properties of a custom currency
```
Comment (monitz87) [inline src/types/index.ts]: Custom currency in which a Tiered STO can raise funds

File: src/entities/TieredSto.ts [hunk 22]
```diff
@@ -18,13 +18,10 @@ import { Investment } from './Investment';
...
 const { weiToValue } = conversionUtils;
 
-/**
- * Represents a unique sto
- */
 export { UniqueIdentifiers };
 
 /**
- * Represents a tier in the tiered sto
+ * Unique properties of a tier in the tiered sto
```
Comment (monitz87) [inline src/entities/TieredSto.ts]: The name is descriptive enough, no need for this comment

**Comment by victorwiebe** [comment]
Have made necessary changes

File: src/entities/Sto.ts [hunk 23]
```diff
@@ -12,74 +12,175 @@ import {
...
+   */
   public capReached: boolean;
 
+  /**
+   * whether the STO has been finalized or not
+   */
   public isFinalized: boolean;
 
+  /**
+   * whether all tokens due to be sold are issued when the STO starts. If false, the appropriate amount of tokens are issued to the buyer whenever a sale is made
```
Comment (monitz87) [inline src/entities/Sto.ts]:
```suggestion
   * whether all tokens due to be sold are issued when the STO starts. If false, the appropriate amount of tokens is issued to the buyer whenever a sale is made
```

**Pull request merged**